esphome/dashboard/dashboard.py

```diff
@@ -693,20 +693,39 @@ def get(self):
         self.write(json.dumps(sd))
 
 
+def _get_platform_boards(platform, title=None):
+    from esphome.components.esp32.boards import BOARDS as ESP32_BOARDS
+    from esphome.components.esp8266.boards import BOARDS as ESP8266_BOARDS
+    from esphome.components.rp2040.boards import BOARDS as RP2040_BOARDS
+
+    boards = {
+        "esp32": ESP32_BOARDS,
+        "esp8266": ESP8266_BOARDS,
+        "rp2040": RP2040_BOARDS,
+    }
+
+    is_esp32 = platform.startswith("esp32")
+    platform_boards = boards["esp32"] if is_esp32 else boards[platform]
+    variant = platform.upper() if is_esp32 else None
+
+    boards_items = {
+        key: val[const.KEY_NAME]
+        for key, val in platform_boards.items()
+        if variant is None or val[const.KEY_VARIANT] == variant
+    }
+    boards_items = sorted(boards_items.items(), key=lambda item: item[1])
+    return {
+        platform: {
+            "items": dict(boards_items),
```

Comment:
this is already a dict.
```suggestion
            "items": boards_items,
```

Reply:
Why wrap everything in an `"items"` ?

Reply:
To allow defining board groups (like the vendor grouping, in LibreTuya for example). I know it's not currently used, but sooner or later I'll need this functionality.

> this is already a dict.

No, `boards_items` is a list of tuples. I'll refactor this further to make variable names less ambiguous.